### ISSUE TITLE
Mochawesome and JUnit reporters are reporting inconsistently

### ISSUE THREAD

**Issue by CBonade** [department-of-veterans-affairs/va.gov-team#27140]
We have two reporting mechanisms in our GHA workflow - the JUnit reporter that spits out the summary in it's own workflow step, and the Mochawesome reports that compile and currently push to Github Pages. Even though the tests only run once, they generate separate sets of results.  The JUnit results are xml that is generated by Cypress, while the Mochawesome combines json partials to create the results.  In multiple runs, the number of "total tests" that have run are differing.  On some occasions there's more in Mochawesome, and in other occasions there are less.

One place to start troubleshooting may be that the json partials are auto-named by their timestamp.  Due to parallelization, it seems that it may be possible that we wind up with multiple results that have the same name, causing some to drop out. A possible solution here, could be to change the naming convention to crutch on something like the parallelization index passed in from Github Actions as an environmental variable and see if it straightens it out at all.

https://github.com/adamgruber/mochawesome-report-generator#options docs on the mochawesome reporter, particularly the option `reportFilename`

**Comment by CBonade** [comment]
Took a quick look at this today before shifting to another priority.  So far, it seems the gap has become more consistent as 3 runs in a row produced the exact same difference between the JUnit reporter and the Mochawesome report. I tested the above hypothesis that changing the partials names may prevent duplicates but it so far did not seem to impact the results.

**Comment by CBonade** [comment]
Did a lot of digging into this today.  It seems that the root cause is somewhere in Cypress where it's handling generating XML results (required for the JUnit reporter) and JSON results (required for mochawesome) differently and coming up with different results.  My current course of action is to try and eliminate the JUnit reporter entirely by generating our own summary based on the information we already have in the mochawesome report.  More to come soon.

**Comment by CBonade** [comment]
The custom summary has been created and it now uses the same set of results as the Mochawesome report to get the numbers. This will ensure that going forward the numbers always match. Additionally, I was able to remove an artifact upload/download that is no longer needed, as well as removing the need for Cypress to generate two sets of results (xml results have been removed from the config).  This PR is set for review.

![image.png](https://images.zenhubusercontent.com/60aa993146d844d86f494162/4937219d-ca58-4a9a-82c9-e7e6a8e64bd9)
